gutenberg/blocks/ts/src/electronic-service-channel-block.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { wp, WPBlockTypeEditParams } from 'wp';
+import Icon from "./electronic-service-channel-icon";
+import ElectronicServiceChannelComponent from './components/electronic-service-channel-component';
+import { SptvOptions } from './types';
+
+declare var wp: wp;
+declare var sptv: SptvOptions;
+const { __ } = wp.i18n;
+
+const { registerBlockType } = wp.blocks;
+
+/**
+ * Registers block type
+ */
+registerBlockType('sptv/electronic-service-channel-block', {
+  title: __( 'Electronic service channel', 'sptv' ),
+  icon: Icon,
+  category: 'sptv',
+
+  attributes: {
+    "id": {
+      type: 'string'
+    },
+    "component": {
+      type: 'string'
+    },
+    "language": {
+      type: 'string'
+    }
+  },
+
+  /**
+   * Block type edit method 
+   */
+  edit: ((params: WPBlockTypeEditParams) => {
+    const { isSelected } = params;
+
+    const getAttribute = (attribute: string): string => {
+      return params.attributes[attribute];
+    }
+
+    const setAttribute = (attribute: string, value: string) => {
+      const attributes: { [key: string]: string } = { }; 
+      attributes[attribute] = value; 
+      params.setAttributes(attributes);
+    }
+
+    if (!getAttribute("component")) {
+      setAttribute("component", sptv.electronicServiceChannelBlock.components ? sptv.electronicServiceChannelBlock.components[0].slug : null);
+    }
+
+    if (!getAttribute("language")) {
+      setAttribute("language", "fi");
+    }
+
+    return <ElectronicServiceChannelComponent 
+      editing={ isSelected } 
+      channelId={ getAttribute("id") } 
+      component={ getAttribute("component") } 
+      language={ getAttribute("language") }
+      onComponentChange={(component: string) => {
+        setAttribute("component", component);
+      }}
+      onLanguageChange={(language: string) => {
+        setAttribute("language", language);
+      }}
+      onChannelIdChange={(channelId: string) => {
+        setAttribute("id", channelId);
+      }}
+      />
+  }),
+
+  /**
+   * Block type save method 
+   */
+  save: (): any => {
+    return null;
+  }
```

Comment:
Does this work as intended?

Reply:
It was there before in other methods, so  suppose it should stay for now